pkg/mcs/exportspec.go

```diff
@@ -0,0 +1,161 @@
+package mcs
+
+import (
+	"bytes"
+	"encoding/json"
+	"os"
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	mcsv1a1 "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+)
+
+// ExprtSpec holds the relevant data for creating imports and arbitrating conflicts
+// in service definitions.
+type ExportSpec struct {
+	// CreatedAt is the API master assigned creation time for the ServiceExport, used in
+	// conflict resolution.
+	CreatedAt metav1.Time `json:"createdAt"`
+	// ClusterID indicates the origin cluster represented by the ServiceExport.
+	ClusterID string `json:"clusterID"`
+	// Namespace indicates the original ServiceExport object namespace.
+	Namespace string `json:"ns"`
+	// Name indicates the original ServiceExport object name.
+	Name string `json:"name"`
+	// Service includes the global properties of the exported Service
+	Service ServiceSpec `json:"spec"`
+}
+
+// ServiceSpec holds the global Service properties as defined by KEP-1645.
+// See https://github.com/kubernetes/enhancements/tree/master/keps/sig-multicluster/1645-multi-cluster-services-api#global-properties.
+type ServiceSpec struct {
+	// Type set for Service (one of ClusterSetIP or Headless).
+	Type mcsv1a1.ServiceImportType `json:"type"`
+	// SessionAffinity set for the Service.
+	SessionAffinity corev1.ServiceAffinity `json:"affinity,omitempty"`
+	// SessionAffinityConfig set for the Service.
+	SessionAffinityConfig *corev1.SessionAffinityConfig `json:"affinityConfig,omitempty"`
+	// Ports defined for the Service.
+	Ports []mcsv1a1.ServicePort `json:"ports"`
+}
+
+const (
+	prefix                  = "lighthouse.submariner.io"
+	ServiceExportAnnotation = prefix + "/" + "serviceExportSpec"
+)
+
+// NewExportSpec creates a new ExportSpec based on the given Service and
+// ServiceExport.
+func NewExportSpec(svc *corev1.Service, export *mcsv1a1.ServiceExport,
+	cluster string) (*ExportSpec, error) {
+
+	if svc == nil || export == nil || cluster == "" {
```

Comment:
Would be useful also to validate that NS and name of export and svc matches.

Reply:
agree. Will add.